docs/lesson-plans/week-4.md

```diff
@@ -0,0 +1,623 @@
+Here are a few of the common ones:
+
+| Code | Name | Description |
+| :--- | :--- | :--- |
+| 200 | OK | The most common one, indicates that requested resource is in or described in the body |
+| 201 | Created | A new resource has been created on the server |
+| 204 | No Content | The request was a success but no body was returned (necessary data *might* be in the headers) |
+
+##### Redirect Responses
+
+Indicates that a resource has been moved or that the client should instead go to a different location.
+
+Here are a few of the common ones:
+
+| Code | Name | Description |
+| :--- | :--- | :--- |
+| 301 | Moved Permanently | The URL has been changed **permanently** and is provided in the response |
+| 302 | Found | The URL has been changed **temporarily** and might be changed again in the future (so keep using this to find it) | 
+
+##### Client Errors
+
+These represent errors that a client has made and are used to let it know what went wrong.
+
+Here are a few of the common ones:
+
+| Code | Name | Description |
+| :--- | :--- | :--- |
+| 400 | Bad Request | There was an issue with the request (usually due to invalid syntax) | 
+| 401 | Unauthorized | The client needs to authenticate (log in) before using this resource |
+| 403 | Forbidden | The client is authorized, but doesn't have the right permissions |
+| 404 | Not Found | The requested resource could not be found or the URL is not recognized |
+| 405 | Method Not Allowed | The wrong method is being used for a route (like a POST vs a GET, will be explained later) |
+| 410 | Gone | The requested resource has been permanently deleted from the server |
+| 418 | I'm a teapot | The server refuses the attempt to brew coffee (yes this is real) | 
+
+##### Server Errors
+
+These responses let the client know that there was an issue processing the request by the server. Usually you want to 
+avoid returning these because you properly handle every error that can pop up, but there are times when it's unavoidable 
+or necessary.
+
+Here are a few of the common ones:
+
+| Code | Name | Description |
+| :--- | :--- | :--- |
+| 500 | Internal Server Error | Usually thrown on an exception the server doesn't know how to handle, you want to try and avoid these |
+| 502 | Bad Gateway | The server (acting as a gateway) got an invalid response (usually this happens for certain server configuration errors) |
+| 503 | Service Unavailable | The server isn't ready to handle the request (usually when the server is down or overloaded) |
+| 504 | Gateway Timeout | When the server (acting as a gateway) could not get a response in a specified timeframe |
+
+#### JSONs
+
+JavaScript Object Notation (JSON) is a way of representing data in a very clean and easy to write way. It's commonly 
+compared to XML. You can see an in-depth description of everything on the 
+[JSON website](https://www.json.org/json-en.html).
+
+The JSON data format consists of three main data types; the `value`, `array`, and `object` types.
+
```

Comment:
You should mention that JSON is very similar to data in JavaScript and there are only a few small syntactical differences.